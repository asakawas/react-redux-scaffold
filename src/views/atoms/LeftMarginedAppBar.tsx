import React from 'react';
import AppBar, { AppBarProps } from '@material-ui/core/AppBar';
import styled from 'styled-components';

/**
 * The type of props of LeftMarginedAppBar.
 */
export type LeftMarginedAppBarProps = Readonly<
  AppBarProps & {
    /** The left margin (in px) of the app bar. */
    marginLeft: number;
  }
>;

const LeftMarginedAppBar = styled(({ marginLeft, ...props }: LeftMarginedAppBarProps) => (
  // Remove LeftMarginedAppBarProps to suppress a runtime warning from react-dom.
  // eslint-disable-next-line react/jsx-props-no-spreading
  <AppBar {...props} />
))`
  margin-left: ${({ marginLeft }: LeftMarginedAppBarProps) => marginLeft}px;
  width: calc(100% - ${({ marginLeft }: LeftMarginedAppBarProps) => marginLeft}px);
`;

export default LeftMarginedAppBar;
