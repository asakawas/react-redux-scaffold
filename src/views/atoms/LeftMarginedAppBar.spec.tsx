import React from 'react';
import { render } from '@testing-library/react';
import LeftMarginedAppBar from './LeftMarginedAppBar';

describe('LeftMarginedAppBar', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<LeftMarginedAppBar marginLeft={200} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('has specified margin in its style attribute', () => {
    const renderResult = render(<LeftMarginedAppBar marginLeft={123} data-testid="test" />);

    const tree = renderResult.getByTestId('test');
    expect(tree).toHaveStyleRule('margin-left', '123px');
    expect(tree).toHaveStyleRule('width', 'calc(100% - 123px)');
  });
});
