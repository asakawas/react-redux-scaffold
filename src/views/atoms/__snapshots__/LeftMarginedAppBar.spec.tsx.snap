// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftMarginedAppBar is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  margin-left: 200px;
  width: calc(100% - 200px);
}

<header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary c0 mui-fixed MuiPaper-elevation4"
  />
</DocumentFragment>
`;
